src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -56,45 +55,50 @@ public class PackageSourceProvider : IPackageSourceProvider
             _configurationDefaultSources = LoadConfigurationDefaultSources(configurationDefaultSources);
         }
 
-        private static IEnumerable<PackageSource> LoadConfigurationDefaultSources(IEnumerable<PackageSource> configurationDefaultSources)
+        private static IReadOnlyList<PackageSource> LoadConfigurationDefaultSources(IEnumerable<PackageSource> configurationDefaultSources)
         {
 #if !IS_CORECLR
             // Global default NuGet source doesn't make sense on Mono
-            if (RuntimeEnvironmentHelper.IsMono)
+            if (Common.RuntimeEnvironmentHelper.IsMono)
             {
-                return Enumerable.Empty<PackageSource>();
+                return Array.Empty<PackageSource>();
             }
 #endif
             var packageSourceLookup = new Dictionary<string, IndexedPackageSource>(StringComparer.OrdinalIgnoreCase);
             var packageIndex = 0;
 
             foreach (var packageSource in configurationDefaultSources)
```

Comment:
I understand this is beyond the scope of this PR. However, if you're addressing other feedback within this PR, please consider incorporating ``.NoAllocEnumerate()``. This addition will help avoid memory allocations if the underlying collection is a list.

```suggestion
            foreach (var packageSource in configurationDefaultSources.NoAllocEnumerate())
```